### DISCUSSION TITLE
React Router's default error element is showing above my error element

### DISCUSSION THREAD

**Discussion by vijtheveg** [remix-run/react-router#11700] [Q&A] [answered]
A strange issue is happening in my ReactJS app. If an exception is thrown when one of my non-root-level components is rendering, the custom error element that I have specified at the root level is shown (```<CatchAllErrorHandler />```), but then React Router's default error element is shown above it! If this default error element is dismissed (by clicking on the 'x' at the top right) my error element is below.

This is what the routes look like:
![image](https://github.com/remix-run/react-router/assets/8265580/1245c0f8-a289-421a-bcf7-af3c96b54a9b)

This is the result when the error is thrown:
<img width="280" alt="Default" src="https://github.com/remix-run/react-router/assets/8265580/f73978b4-9e48-483f-8cd4-86d806e09689">
If you look carefully, you can see the outline of my custom error element under the default error element.

And sure enough, if I dismiss the default error element, my error element is below.
<img width="281" alt="Custom" src="https://github.com/remix-run/react-router/assets/8265580/3aca2221-c58b-45d1-a835-098f9ab1e985">

By the way, in the console, there is a log entry that says
> React will try to recreate this component tree from scratch using the error boundary you provided, RenderErrorBoundary.

Any idea what may be happening? I'll really appreciate any pointers - I have spent hours on this and am none the wiser.

**Comment by tatsmaki** [comment, marked as answer]
This is **react-router** error window
![Screenshot 2024-06-21 at 23 06 28](https://github.com/remix-run/react-router/assets/61022192/cf5d5e0b-85ce-4b32-a847-205cc4e13219)

And this is **webpack** error window (or vite)
![Screenshot 2024-06-21 at 23 07 05](https://github.com/remix-run/react-router/assets/61022192/5cdca64a-9cf5-434b-af56-6f9856c172cb)

This error won't be visible in production. If you want to adjust it for development – check webpack docs to configure devServer overlay
https://webpack.js.org/configuration/dev-server/#overlay

Something like this will work
```
devServer: {
    client: {
         overlay: {
             errors: false
         }
    }
}
```

**Reply by tatsmaki** [reply]
If you are using create-react-app – you can either
1) switch to webpack/vite as soon as possible
2) execute eject script
3) try tools like https://www.npmjs.com/package/customize-cra or https://www.npmjs.com/package/react-app-rewired

**Reply by vijtheveg** [reply]
@tatsmaki Thank you very much for the quick and detailed response - you have been very helpful! It is good to know that this overlay won't appear in production.
